### PR TITLE
Add an output option to qcheck-stm plugin cli

### PR DESCRIPTION
**Pull request by n-osborne**
- Update the pinned gospel version
- Support patterns as arguments of anonymous functions
- Add output option to qcheck-stm plugin cli
- Use the new output option in qcheck-stm tests

### CODE REVIEW

File: plugins/qcheck-stm/src/ortac_qcheck_stm.ml
Comment (shym) [inline plugins/qcheck-stm/src/ortac_qcheck_stm.ml:12]: Why the extra parentheses? (they were there before, I know, but still)

File: plugins/qcheck-stm/src/ortac_qcheck_stm.ml
Comment (shym) [inline plugins/qcheck-stm/src/ortac_qcheck_stm.ml:10]: While reviewing this, I notice that all plugins are actually using a formatter. Maybe `Registration` could provide a `get_out_formatter` rather than a `get_channel`?

**Review by shym — APPROVED** [review]
This is good to go after a reformatting of the `dune` files.
I’ve added minor remarks that could be addressed but feel free to ignore them.

File: plugins/qcheck-stm/src/ortac_qcheck_stm.ml
Comment (shym) [reply plugins/qcheck-stm/src/ortac_qcheck_stm.ml:10]: By the way, I find a formatter called `channel` a bit misleading...

File: plugins/qcheck-stm/src/ortac_qcheck_stm.ml
Comment (n-osborne) [reply plugins/qcheck-stm/src/ortac_qcheck_stm.ml]: No reason, I've removed them.

File: plugins/qcheck-stm/src/ortac_qcheck_stm.ml
Comment (n-osborne) [reply plugins/qcheck-stm/src/ortac_qcheck_stm.ml]: `get_out_formatter` is a good idea.
Added in a third commit to the PR.

**Comment by n-osborne** [comment]
The PR is based on `shym/integration` (part of #136).
So I'll wait for the branch to be merged into main before merging this PR.

**Comment by n-osborne** [comment]
Rebased on main now that #136 is merged

**Comment by shym** [comment]
Close #117

**Comment by shym** [comment]
The failure in CI is unrelated to this PR, I merge.

**Pull request merged**
